test/compiler/inference.jl

```diff
@@ -627,7 +627,7 @@ function maybe_vararg_tuple_1()
 end
 @test Type{Tuple{Vararg{Int}}} <: Base.return_types(maybe_vararg_tuple_1, ())[1]
 function maybe_vararg_tuple_2()
-    x = Type[Vararg{Int}][1]
+    x = [Vararg{Int}][1]
```

Comment:
```suggestion
    x = Any[Vararg{Int}][1]
```

Reply:
This is already tested by the test just above. I figured the most natural translation would be to just have this be typed as ` typeof(Vararg)`..